### ISSUE TITLE
[Bug]: `aws_securitylake_aws_log_source` source version is optional, but returns error

### ISSUE THREAD

**Issue by gdavison** [hashicorp/terraform-provider-aws#37133]
### Terraform Core Version

N/A

### AWS Provider Version

5.47.0

### Affected Resource(s)

aws_securitylake_aws_log_source

### Expected Behavior

When using a default `source_version`, creating the `aws_securitylake_aws_log_source` should succeed.

### Actual Behavior

Applying the configuration returns the error

> Error: Provider returned invalid result object after apply
        
        After the apply operation, the provider still indicated an unknown value for
        aws_securitylake_aws_log_source.test.source[0].source_version. All values
        must be known after apply, so this is always a bug in the provider and should
        be reported in the provider's own repository. Terraform will still save the
        other known object values in the state.

### Terraform Configuration Files

N/A

### Steps to Reproduce

N/A

### Would you like to implement a fix?

None

**Comment by rl-gr24** [comment]
Facing a similar issue with log source. got it to work under specific circumstances/parameters. A few observations below:

Observation 1: [ERROR] ran the following through the delegated securitylake admin account and got the same error as the issue description. But when i check the console, the logs are indeed enabled in all accounts in the org. This org has accounts that are currently in suspended state.
resource "aws_securitylake_aws_log_source" "route53" {
    source {
        regions     = ["eu-west-1"]   ###includes suspended and active accounts in the organization ####
        source_name = "ROUTE53"
    }
  
    depends_on = [aws_securitylake_data_lake.securitylake]
}

Observation 2: [ERROR] Destroyed previous run w/ TF and ran the following through the delegated securitylake admin account and got the same error again. The only difference is that I added  **source_version = "2.0"**
resource "aws_securitylake_aws_log_source" "route53" {
    source {
        regions     = ["eu-west-1"]   ###includes suspended and active accounts in the organization ####
        source_name = "ROUTE53"
     **source_version = "2.0"  ###Added source versioning here###**
    }
  
    depends_on = [aws_securitylake_data_lake.securitylake]
}

Observation 3: [SUCCESSFUL]
Destroyed the previous log sources and then ran the following through the delegatedadmin account again with  two additions below and was SUCCESSFUL
resource "aws_securitylake_aws_log_source" "vpc_flow" {
    source {
     **account    = ["active_account_ID1", "active_account_ID2", "..." ] ##Ensured no suspended account IDs were included##**
        regions     = ["eu-west-1"]
        source_name = "ROUTE53"
     **source_version = "2.0"  ###Added source versioning here###**
    }
    depends_on = [aws_securitylake_data_lake.securitylake]
}

**Comment by rl-gr24** [comment]
Obviously, hardcoding account ID is not ideal, especially in a large and growing AWS organization with many accounts.
